spec.html

```diff
@@ -12,194 +12,174 @@ <h1>Locale Objects</h1>
+          1. Let _subtag_ the String value equal to the substring of _extension_ consisting of the code units at indices _k_ (inclusive) through _k_ + _len_ (exclusive).
+          1. If _len_ = 2, then
+            1. If _keywords_ does not contain an element whose [[Key]] is the same as _key_, then
+              1. Append the Record{[[Key]]: _key_, [[Value]]: _value_} to _keywords_.
+            1. Let _key_ be _subtag_.
+            1. Let _value_ be the empty String.
+          1. Else,
+            1. If _value_ is not the empty String, then
+              1. Let _value_ be the string-concatenation of _value_ and `"-"`.
+            1. Let _value_ be the string-concatenation of _value_ and _subtag_.
+          1. Let _k_ be _k_ + _len_ + 1.
+        1. If _keywords_ does not contain an element whose [[Key]] is the same as _key_, then
+          1. Append the Record{[[Key]]: _key_, [[Value]]: _value_} to _keywords_.
+        1. Let _result_ be a new Record.
+        1. Let _newExtension_ be `"-u"`.
+        1. Let _entry_ be the element of _keywords_ whose [[Key]] is the same as `"attributes"`.
+        1. If _entry_.[[Value]] is not the empty String, then
+          1. Let _newExtension_ be the string-concatenation of _newExtension_, `"-"`, and _entry_.[[Value]].
+        1. Remove _entry_ from _keywords_.
+        1. Repeat for each element _key_ of _relevantExtensionKeys_ in List order,
+          1. Let _value_ be *undefined*.
+          1. If _keywords_ contains an element whose [[Key]] is the same as _key_, then
+            1. Let _entry_ be the element of _keywords_ whose [[Key]] is the same as _key_.
+            1. Let _value_ be _entry_.[[Value]].
+            1. Remove _entry_ from _keywords_.
+          1. Assert: _options_ has a field [[<_key_>]].
+          1. Let _optionsValue_ be _options_.[[<_key_>]].
+          1. If _optionsValue_ is not *undefined*, then
+            1. Let _value_ be _optionsValue_.
+          1. Set _result_.[[<_key_>]] to _value_.
+          1. If _value_  is not *undefined*, then
+            1. Assert: Type(_value_) is String.
+            1. Let _newExtension_ be the string-concatenation of _newExtension_, `"-"`, and _key_.
+            1. If _value_ is not the empty String, then
+              1. Let _newExtension_ be the string-concatenation of _newExtension_, `"-"`, and _value_.
+        1. Repeat for each element _entry_ of _keywords_ in List order,
+          1. Let _key_ be _entry_.[[Key]].
+          1. Let _value_ be _entry_.[[Value]].
+          1. Let _newExtension_ be the string-concatenation of _newExtension_, `"-"`, and _key_.
+          1. If _value_ is not the empty String, then
+            1. Let _newExtension_ be the string-concatenation of _newExtension_, `"-"`, and _value_.
+        1. Let _locale_ be the String value that is _tag_ with all Unicode locale extension sequences removed.
+        1. If the number of elements in _newExtension_ is greater than 2, then
```

Comment:
Are there any cases where there is a Unicode extension sequence but nothing gets appended to _newExtension_? If not, I think the preceding section would be a bit clearer if this conditional were hoisted up to line 64 (namely, maybe the else case should skip straight down to line 126).

Reply:
> (namely, maybe the else case should skip straight down to line 126).

Do you mean the `Else` at line 66? If we go from 66 directly 126 (well, actually 128), we'd skip the loop at line 95, which we need to run even if no Unicode extension sequences are present, because it appends Unicode extension keywords from the `options` into the result locale.

Lines 68-88 could be moved into a separate abstract operation, so it can also be used in `ResolveLocale` and replace the existing `UnicodeExtensionValue` operation.

Reply:
I see, never mind.